### ISSUE TITLE
Tips for avoiding exploding gradients in the generator?

### ISSUE THREAD

**Issue by usersub0** [affinelayer/pix2pix-tensorflow#20]
When training, I'm getting NaN errors on some layers. The actual layer depends on my seed but, for example, I just got the error :
```
Nan in summary histogram for: generator/decoder_7/batchnorm/offset/values
```
Sometimes I'll also get them on the encoder gradients. At least in those cases, inspecting the corresponding gradients via tensorboard/histograms shows huge values. 

Did you come across this at all and any tips for how to adjust for these?

I'm training for colorization on my own dataset. The error tend to happen within the first 2000 images. Viewing the resulting generated imaged in tensorboard up to that point show promising results. I've played around with varying the 0.2 parameter getting used in the leaky relu layers but am not seeing any obvious improvements.

**Comment by christopherhesse** [comment]
I haven't seen this problem myself, but you can try clipping the gradients with something like this: grads = [tf.clip_by_value(g, -5, 5) for g in grads]

**Comment by WLpub** [comment]
Hi @usersub0 , You can check wgan_gp, which provide better way than weight clips referenced by @christopherhesse

**Comment by yanfeiwong** [comment]
Have you solved it? Im running into the same issue with colorization ...
